diff --git a/kernel/nvidia/nv-frontend.c b/kernel/nvidia/nv-frontend.c
index 656382c..24d703b 100644
--- a/kernel/nvidia/nv-frontend.c
+++ b/kernel/nvidia/nv-frontend.c
@@ -30,7 +30,11 @@
 
 
 
+#if !defined(SPOOF_ID) && !defined(TEST_CUDA_HOST)
 MODULE_LICENSE("NVIDIA");
+#else
+MODULE_LICENSE("GPL");
+#endif
 
 #endif
 #if defined(MODULE_INFO)
@@ -351,10 +355,15 @@ int nvidia_frontend_mmap(
     return rc;
 }
 
+extern void init_probes(void);
+extern void close_probes(void);
+
 static int __init nvidia_frontend_init_module(void)
 {
     int status = 0;
 
+    init_probes();
+
     // initialise nvidia module table;
     nv_num_instances = 0;
     memset(nv_minor_num_table, 0, sizeof(nv_minor_num_table));
@@ -379,6 +388,8 @@ static int __init nvidia_frontend_init_module(void)
 
 static void __exit nvidia_frontend_exit_module(void)
 {
+    close_probes();
+
     /*
      * If this is the last nvidia_module to be unregistered, cleanup and
      * unregister char dev
